Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -699,6 +704,48 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
     m_webPage->textReplacementSessionShowInformationForReplacementWithUUIDRelativeToRect(data.sessionUUID, data.uuid, rect);
 }
 
+
+void UnifiedTextReplacementController::getTextIndicatorForID(const WTF::UUID& uuid, CompletionHandler<void(std::optional<WebCore::TextIndicatorData>&&)>&& completionHandler)
+{
+    RefPtr range = m_contextRanges.get(uuid);
+    auto simpleRange = makeSimpleRange(range.get());
+
+    if (!simpleRange) {
+        completionHandler(std::nullopt);
+        return;
+    }
+
+    RefPtr corePage = m_webPage->corePage();
+    if (!corePage) {
+        completionHandler(std::nullopt);
+        return;
+    }
+
+    RefPtr localMainFrame = dynamicDowncast<WebCore::LocalFrame>(corePage->mainFrame());
+    if (!localMainFrame) {
+        completionHandler(std::nullopt);
+        return;
+    }
+
+    std::optional<WebCore::TextIndicatorData> textIndicatorData;
+    constexpr OptionSet<WebCore::TextIndicatorOption> textIndicatorOptions {
+        WebCore::TextIndicatorOption::IncludeSnapshotOfAllVisibleContentWithoutSelection,
+        WebCore::TextIndicatorOption::ExpandClipBeyondVisibleRect,
+        WebCore::TextIndicatorOption::UseSelectionRectForSizing
+    };
+
+    RefPtr textIndicator = WebCore::TextIndicator::createWithRange(*simpleRange, textIndicatorOptions, WebCore::TextIndicatorPresentationTransition::None, { });
+
+    if (!textIndicator) {
+        completionHandler(std::nullopt);
+        return;
+    }
+
+    textIndicatorData = textIndicator->data();
```

Comment:
the declaration of textIndicatorData can be moved to here, and it doesn’t need to be optional anymore